source/adapters/cuda/enqueue.cpp

```diff
@@ -484,6 +484,133 @@ UR_APIEXPORT ur_result_t UR_APICALL urEnqueueCooperativeKernelLaunchExp(
                                numEventsInWaitList, phEventWaitList, phEvent);
 }
 
+UR_APIEXPORT ur_result_t UR_APICALL urEnqueueKernelLaunchCustomExp(
+    ur_queue_handle_t hQueue, ur_kernel_handle_t hKernel, uint32_t workDim,
+    const size_t *pGlobalWorkSize, const size_t *pLocalWorkSize,
+    uint32_t numAttrsInLaunchAttrList,
+    const ur_exp_launch_attribute_t *launchAttrList,
+    uint32_t numEventsInWaitList, const ur_event_handle_t *phEventWaitList,
+    ur_event_handle_t *phEvent) {
+
+  if (numAttrsInLaunchAttrList == 0) {
+    urEnqueueKernelLaunch(hQueue, hKernel, workDim, nullptr, pGlobalWorkSize,
+                          pLocalWorkSize, numEventsInWaitList, phEventWaitList,
+                          phEvent);
+  }
+
+  // Preconditions
+  UR_ASSERT(hQueue->getContext() == hKernel->getContext(),
```

Comment:
Nit: I'd be more concerned that `hQueue->getDevice() == hKernel->getProgram()->getDevice()`

Reply:
Thanks, I forgot about the multi-context patch. I've now made this consistent with it.